sros2/scripts/dds_sql_to_sros2_policy.py

```diff
@@ -0,0 +1,172 @@
+
+# coding: utf-8
+
+# In[ ]:
+
+
+import pandas as pd
+import sqlite3
+
+from lxml import etree
+
+
+# In[ ]:
+
+
+db_path = "../tb3_demo_dds_discovery.db"
+
+
+# In[ ]:
+
+
+node_pub_query = """
+SELECT DISTINCT
+    DCPSParticipant.ParticipantData_key,
+    DCPSParticipant.ParticipantData_user_data,
+    DCPSPublication.PublicationData_topic_name
+FROM DCPSParticipant 
+INNER JOIN DCPSPublication ON
+    DCPSParticipant.ParticipantData_key = DCPSPublication.PublicationData_participant_key
+AND DCPSParticipant.ParticipantData_user_data IS NOT NULL;
+"""
+
+node_sub_query = """
+SELECT DISTINCT
+    DCPSParticipant.ParticipantData_key,
+    DCPSParticipant.ParticipantData_user_data,
+    DCPSSubscription.SubscriptionData_topic_name
+FROM DCPSParticipant 
+INNER JOIN DCPSSubscription ON
+    DCPSParticipant.ParticipantData_key=DCPSSubscription.SubscriptionData_participant_key
+AND DCPSParticipant.ParticipantData_user_data IS NOT NULL;
+"""
+
+
+# In[ ]:
+
+
+def user_bytes_to_dict(user_bytes):
+    user_string = user_bytes.decode('utf8')
+    key_value_list = user_string[:-2].split(';')
+    key_value_dict = dict()
+    for key_values in key_value_list:
+        key, value = key_values.split('=', 1)
+        key_value_dict[key] = value
+    return key_value_dict
+
+
+# In[ ]:
+
+
+def translate_df(df):
+    _df = df['ParticipantData_user_data'].apply(user_bytes_to_dict).apply(pd.Series)
+    df = pd.concat([df, _df], axis=1)
+    return df
+
+
+# In[ ]:
+
+
+db_uri = 'file:{}?mode=ro'.format(db_path)
+with sqlite3.connect(db_uri, uri=True) as db:
+    pub_df = translate_df(pd.read_sql_query(node_pub_query, db))
+    sub_df = translate_df(pd.read_sql_query(node_sub_query, db))
+
+pub_df = pub_df.assign(mode='publish')
+pub_df = pub_df.rename(columns={"PublicationData_topic_name": "dds_topic"})
+sub_df = sub_df.assign(mode='subscribe')
+sub_df = sub_df.rename(columns={"SubscriptionData_topic_name": "dds_topic"})
+
+df = pd.concat([pub_df, sub_df])
+df.set_index(['namespace', 'name', 'mode'], inplace=True)
+
+
+# In[ ]:
+
+
+foo = df.index[0]
+foo
+
+
+# In[ ]:
+
+
+# with pd.option_context('display.max_rows', None, 'display.max_columns', None):  # more options can be specified also
+#     display(df)
+
+
+# In[ ]:
+
+
+polciy = etree.Element('polciy')
+profiles = etree.SubElement(polciy, 'profiles')
+
+for namespace in df.index.get_level_values('namespace').unique():
+    for name in df.index.get_level_values('name').unique():
+        profile = etree.SubElement(profiles, 'profile')
+        profile.set("ns", namespace)
+        profile.set("node", name)
+        _df = df['dds_topic'].loc[namespace, name, :]
+        for mode in _df.index.get_level_values('mode').unique():
+            topics = etree.SubElement(profile, 'dds_topics')
+            topics.set(mode, "ALLOW")
+            for dds_topic in df['dds_topic'].loc[namespace, name, :]:
```

Comment:
I believe this should check for `mode` to not add all topics corresponding to this namespace + name